src/api/app/models/event/workflow_run_fail.rb

```diff
@@ -0,0 +1,17 @@
+module Event
+  class WorkflowRunFail < Base
+    self.message_bus_routing_key = 'workflow_run.fail'
+    self.description = 'Workflow run has failed'
+    payload_keys :id, :token_id # TODO: :generic_event_type, :hook_action, :hook_event, :repository_name, :repository_owner, :scm_vendor
+
+    receiver_roles :token_executor
+
+    def token_executor
```

Comment:
I think the receiver role finder method should be in plural form. See [here](https://github.com/openSUSE/open-build-service/blob/a7112563a0ae9b57ce18e668ff955c0410da003b/src/api/app/models/event_subscription/find_for_event.rb#L16)

Reply:
True! We realized later on.